libbeat/docs/upgrading.asciidoc

```diff
@@ -1,160 +1,329 @@
+with indices created by {beats} 7.0 or later.
 
-===== New command `test config` command
+Did you run the Beat before reading this section? That's OK. See the clean-up
+steps described under <<missing-fields>>.
 
-Beats 6.0 introduces a new test command for testing the configuration file. For
-example:
+[[upgrade-mapping-template]]
+==== Upgrade the {es} mapping template
 
-[source,shell]
+Mapping templates and the default index names are versioned. For example,
+Metricbeat {stack-version} typically creates indices like this:
+
+["source","sh",subs="attributes"]
 ------------------------------------------------------------------------------
-metricbeat test config
+metricbeat-{stack-version}-{localdate}
 ------------------------------------------------------------------------------
 
-The old `-configtest` flag is still available, but deprecated.
+And the corresponding {es} template is named +metricbeat-{stack-version}+.
+
+This means that each version of the Beat creates a new index, and it's
+guaranteed that the correct template for that version is applied. With these
+changes in place, you generally don't have to do anything to upgrade the mapping
+template when you move to a new version. Just load the index template *before*
+ingesting any data into {es}.
 
-===== Reference configuration files
+To load the index template, run the `setup` command with the `--template` option
+specified. For example:
 
-The `<beatname>.full.yml` file, which contains all the non-deprecated
-configuration options is renamed to `<beatname>.reference.yml` starting with
-Beats 6.0. We recommend using this file as a reference only. It's not intended
-to be used in production.
+[source,shell]
+----
+metricbeat setup --template
+----
 
-==== Dashboard upgrades
+If your {beats} configuration use an output other than `elasticsearch`,
+temporarily enable the `elasticsearch` output when you run the `setup` command.
+For example:
 
-We recommend re-importing the Kibana dashboards after the Beats and Kibana
-upgrades are complete. This way, you will take advantage of the new Beats
-dashboards created for the 6.0 release.
+[source,shell]
+--
+metricbeat setup --template -E output.logstash.enabled=false -E 'output.elasticsearch.hosts=["localhost:9200"]'
+--
 
-See <<breaking-changes-import-dashboards>> in the Breaking Changes
-documentation.
+[[non-es-outputs]]
+==== How to use versioned templates when the output is not {es}
 
-NOTE: If you've modified the Beats dashboards manually, reimporting them will
-overwrite your changes. Consider saving your dashboards under new IDs, so
-that the reimport doesn't overwrite them.
+If you've configured {beats} to send events to a different output, such as {ls},
+make sure you use versioned templates and indices. Otherwise, after you
+upgrade, there will be field conflicts. To use versioned index templates and
+indices, use the metadata from {beats} to set the index, and allow {beats} to
+manage the template.
```

Comment:
Can we reorder contexts so that first we discuss this in the context of Logstash and then we extend to the general case? I think this will help people follow along, though maybe I'm just biased because I've been doing all of my testing with Logstash. 😄 

> To use versioned index templates and indices with Logstash, configure your Logstash pipeline to use the metadata from Beats to set the index, and allow Beats to manage the template.
>
> You'll do this by setting the Logstash `index` setting as recommended in the documentation and allowing Beats to manage the index template:
>
> (Example config)
>
> To use versioned index templates and indices with other non-ES output, use this metadata from {beats} to set the index, and allow {beats} to
manage the template.

Reply:
Yes, I think that would work because I suspect this section applies mostly to LS users.